rust/src/storage/s3/mod.rs

```diff
@@ -777,9 +777,14 @@ impl StorageBackend for S3StorageBackend {
         let lock_client = match self.s3_lock_client {
             Some(ref lock_client) => lock_client,
             None => {
-                return Err(StorageError::S3Generic(
-                    "dynamodb locking is not enabled".to_string(),
-                ))
+                /// instead of throw exception, rename in unsafe way. The concurrency control should be done in app side.
+                log::info!(
+                            "Rename from `{}` to {} without s3 lock",
+                            name,
+                            record,
+                        );
```

Comment:
The concurrency description you have is accurate and it would great if we have this included in our docs :)

I agree that the logging here will be too verbose. I think the best/safest thing to do is to require explicit opt-in for lockless s3 support. i.e. if the user wants to use s3 without concurrent writer support, they need to explicitly set a storage backend option. For example, we could add a `single_writer` boolean option to the `S3StorageOptions` struct. That way, we won't need to log this for every call because user already explicitly assumed the risks.